src/test/RUNTESTS.py

```diff
@@ -87,6 +87,10 @@ def run_tests(self):
                     try:
                         t = tc()
                         if t.enabled:
+                            if self.config.list_tests:
```

Comment:
This is a hack. Abstract out test iteration into a separate function and use it both for test execution and listing.